src/Protocol/ProtocolRecognizer.h

```diff
@@ -52,6 +52,9 @@ class cProtocolRecognizer :
 	/** Translates protocol version number into protocol version text: 49 -> "1.4.4" */
 	static AString GetVersionTextFromInt(int a_ProtocolVersion);
 
+	/** GetPacketId is implemented in each protocol version class */
+	virtual UInt32 GetPacketId(eOutgoingPackets a_Packet) override { return 0; }
```

Comment:
It doesn't look like this should ever be reached? Perhaps if so, it is better to fail an assertion or something instead of returning zero.